docs/CONTRIBUTING.md

```diff
@@ -35,6 +35,58 @@ The source code of ABACUS is based on several modules. Under the ABACUS root dir
 - `tests`: End-to-end test cases;
 - `tools`: the script for generating the numerical atomic orbitals.
 
+For those who are interested in the source code, the following figure shows the structure of the source code.
+
+```
+|-- module_base                 A basic module including 
+|   |                           (1) Mathematical library interface functions: BLAS, LAPACK, Scalapack;
+|   |                           (2) Custom data classes: matrix, vector definitions and related functions;
+|   |                           (3) Parallelization functions: MPI, OpenMP;
+|   |                           (4) Utility functions: timer, random number generator, etc.
+|   |                           (5) Global parameters: input parameters, element names, mathematical and physical constants.
+|   |-- module_container        The container module for storing data and performing operations on them and on different architectures.
+|-- module_basis                Basis means the basis set to expand the wave function.
+|   |-- module_ao               Atomic orbital basis set to be refactored.
+|   |-- module_nao              New numerical atomic orbital basis set for two-center integrals in LCAO calculations
+|   `-- module_pw               Plane wave basis set for PW calculations
+|-- module_cell                 The module for defining the unit cell and its operations, and reading pseudopotentials.
+|   |-- module_neighbor         The module for finding the neighbors of each atom in the unit cell.
+|   |-- module_paw              The module for performing PAW calculations.
+|   |-- module_symmetry         The module for finding the symmetry operations of the unit cell.
+|-- module_elecstate            The module for defining the electronic state and its operations.
+|   |-- module_charge           The module for calculating the charge density, charge mixing
+|   |-- potentials              The module for calculating the potentials, including Hartree, exchange-correlation, local pseudopotential, etc.
+|-- module_esolver              The module for solving the eigenvalue problem with different theories including Kohn-Sham DFT, Stochastic DFT,
```

Comment:
to my best present knowledge and my understanding on our codes, the esolver actually acts as a driver, for different esolver_type defined during parsing of INPUT, corresponding workflow will be assigned and carried out. I have noticed in description of hsolver is already "solving the Hamiltonian with different diagonalization methods, including ...", so my suggestion would be changing present description to "the module defining task-specific driver of corresponding workflow, including lj, dp, ks, sdft, etc."